src/components/GlobalFilter/GlobalFilter.tsx

```diff
@@ -1,51 +1,46 @@
 import React, { useCallback, useContext, useEffect, useMemo, useState } from 'react';
-import { batch, shallowEqual, useDispatch, useSelector } from 'react-redux';
 import { useTagsFilter } from '@redhat-cloud-services/frontend-components/FilterHooks';
 import debounce from 'lodash/debounce';
-import { fetchAllSIDs, fetchAllTags, fetchAllWorkloads, globalFilterChange } from '../../redux/actions';
+import { fetchAllSIDs, fetchAllTags, fetchAllWorkloads, globalFilterChange } from '../../state/actions/globalFilterActions';
 import { generateFilter } from './globalFilterApi';
 import { useLocation, useNavigate } from 'react-router-dom';
 import { GlobalFilterDropdown, GlobalFilterDropdownProps } from './GlobalFilterMenu';
 import { storeFilter } from './filterApi';
-import { GlobalFilterTag, GlobalFilterWorkloads, ReduxState, SID } from '../../redux/store';
+import { GlobalFilterTag, GlobalFilterWorkloads, SID } from '../../@types/types';
 import { FlagTagsFilter } from '../../@types/types';
 import { isGlobalFilterAllowed } from '../../utils/common';
 import InternalChromeContext from '../../utils/internalChromeContext';
 import ChromeAuthContext from '../../auth/ChromeAuthContext';
 import { useAtomValue } from 'jotai';
 import { activeModuleAtom } from '../../state/atoms/activeModuleAtom';
+import { globalFilterReducerAtom } from '../../state/atoms/globalFilterAtom';
+import useGlobalFilterState from '../../hooks/useGlobalFilterState';
 
 const useLoadTags = (hasAccess = false) => {
   const navigate = useNavigate();
-  const registeredWith = useSelector(({ globalFilter: { scope } }: ReduxState) => scope);
+  const globalFilterState = useAtomValue(globalFilterReducerAtom);
+  const registeredWith = globalFilterState.scope;
   const activeModule = useAtomValue(activeModuleAtom);
-  const isDisabled = useSelector(({ globalFilter: { globalFilterHidden } }: ReduxState) => globalFilterHidden || !activeModule);
-  const dispatch = useDispatch();
+  const isDisabled = globalFilterState.globalFilterHidden || !activeModule;
   return useCallback(
     debounce((activeTags: any, search: any) => {
       storeFilter(activeTags, hasAccess && !isDisabled, navigate);
-      batch(() => {
-        dispatch(
-          fetchAllTags({
-            registeredWith,
-            activeTags,
-            search,
-          })
-        );
-        dispatch(
-          fetchAllSIDs({
-            registeredWith,
-            activeTags,
-            search,
-          })
-        );
-        dispatch(
-          fetchAllWorkloads({
-            registeredWith,
-            activeTags,
-            search,
-          })
-        );
+      // will removing the batch here cause problems? they are all fetches
```

Comment:
Did we have race conditions here? Curious why these were batched together. In that discussion, somebody else locked the resources before fetching to imitate the batch. Should we do that?

Reply:
You can create a write atom that will set the variables in a single callback. But I do not think removing the batch should be an issue now.